packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -2501,5 +2507,404 @@ describe("Runtime", () => {
+				);
+			});
+
+			it("emits SignalOutOfOrder error event when missing signal is received non-sequentially", () => {
+				sendSignals(3);
+				processSubmittedSignals(1);
+				dropSignals(1);
+				processSubmittedSignals(1);
+
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 1,
+						},
+					],
+					"SignalLost telemetry should be logged when signal is dropped",
+					undefined,
+					false,
+				);
+
+				logger.assertMatchNone(
+					[
+						{
+							eventName: "ContainerRuntime:SignalOutOfOrder",
+						},
+					],
+					"SignalOutOfOrder telemetry should not be logged on lost signal",
+				);
+
+				processDroppedSignals(1);
+
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalOutOfOrder",
+						},
+					],
+					"SignalOutOfOrder telemetry should be logged when missing signal is received non-sequentially",
+				);
+			});
+
+			it("does not emit error events when signals are processed in order", () => {
+				sendSignals(100);
+
+				processSubmittedSignals(100);
+
+				logger.assertMatchNone(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+						},
+						{
+							eventName: "ContainerRuntime:SignalOutOfOrder",
+						},
+					],
+					"SignalLost and SignalOutOfOrder telemetry should not be logged when signals are processed in order",
+				);
+			});
+
+			it("logs relative lost signal count in SignalLost telemetry", () => {
+				sendSignals(5);
+				dropSignals(1);
+				processSubmittedSignals(1);
+
+				// Missing signal should be detected
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 1,
+						},
+					],
+					"SignalLost telemetry should be logged when signal is dropped",
+				);
+
+				dropSignals(2);
+				processSubmittedSignals(1);
+
+				// Missing 3rd and 4th signals should be detected
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 2,
+						},
+					],
+					"SignalLost telemetry should be logged when signal is dropped",
+				);
+			});
+
+			it("ignores in-flight signals on disconnect/reconnect", () => {
+				sendSignals(4);
+
+				// Disconnect + Reconnect
+				changeConnectionState(containerRuntime, false, mockClientId);
+				changeConnectionState(containerRuntime, true, mockClientId);
+
+				// Temporarily lose two old signals
+				dropSignals(2);
+
+				// Receive old signals sent before disconnect
+				processSubmittedSignals(2);
+
+				// Receive old out of order signals
+				processDroppedSignals(2);
+
+				// No error events should be logged for signals sent before disconnect
+				logger.assertMatchNone(
+					[
+						{
+							eventName: "ContainerRuntime:SignalOutOfOrder",
+						},
+						{
+							eventName: "ContainerRuntime:SignalLost",
+						},
+					],
+					"SignalOutOfOrder/SignalLost telemetry should not be logged on reconnect",
+				);
```

Comment:
More effective if there is also a positive test for some telemetry event - post reconnect.

Reply:
added a positive reconnect test after disconnect/reconnect flow that also tests for incremented reconnect count on new SignalLatency event.